docs/draft-hfc-proposal.md

```diff
@@ -0,0 +1,70 @@
+
+- According to Ouroboros Chronos, the ChainSync mini protocol client should disconnect from the upstream peer if the header is from the far future and otherwise briefly pause until it is no longer from the future.
+
+    - (This is the H of HLQ.)
+
+    - The ChainSync client should annotate the validated header with the wall clock translation (eg UTC time) of the header's slot's onset (translated accorded to its own chain).
+      This information can be used by performance monitors, etc.
+
+    - Node initialization from the on-disk block database is the only other means by which blocks arrive at the node, so the re-application/re-validation logic used there should also yield wall clock annotations on the node's selected header chain.
+
+- The node should use its current selection's tip in order to determine whether it is time to mint a block.
+  If the wall clock reaches the onset of a slot before the node determines it leads that slot, the node should not mint a block in that slot.
+  Moreover, after minting a block, the node should not mint again until the wall clock reaches the end of the slot of the block it most recently minted.
+
+    - (This is the L of HLQ.)
+
+    - According to CG and the Praos epoch structure (eg 3:4:4), a node will never switch chains such that the leader schedule changes.
+      Similarly, a chain switch will never change the era of the current epoch.
+
+    - Even so, the above rules suffice to prevent the honest node from equivocating elections when CG is violated.
+
+    - It would be some what simpler to instead only mint according to the leadership schedule of the k+1 youngest block, and the prevention of equivocation is obvious.
+      However, if the node's selection violates CG, the node might not be able to mint: the k+1 youngest block is not necessarily able to translate the wall clock to a slot number in the presence of a CG violation.
+
+- The interface exposed to clients by the node should make it extremely explicit when a subsequent chain switch might change the node's answer to a query (eg which upcoming slots the node leads).
+
+    - (This is the Q of HLQ.)
+
+    - In the extreme, those queries could be restricted to forecasts from the node's k+1 youngest block, thereby ensuring no chain switch could alter the answer except monotonically improving it from "unknown" to "known".
+      In less extreme cases, the documentation could say "you might see non-monotonic answers if using an acquired ledger state that does not have more than k blocks past the voting deadline".
+      A middle ground might be annotating the answer with the corresponding probability from IOG's latest official publication of the table of settlement times.
+
+- In the Praos and Genesis nodes, there is no use of wall clock times beyond the header future check, slot leadership check, and queries discussed above.
```

Comment:
There actually is one in the (legacy) logic for determining which mode BlockFetch shall use: https://github.com/IntersectMBO/ouroboros-consensus/blob/4519035ade017bd1d8399b8027a5147fb2afbb3e/ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/BlockFetch/ClientInterface.hs#L143

This sounds very minor; we can either remove it completely (and just rely on the GSM instead), or replace the current ad-hoc slot different with an ad-hoc wallclock time difference.